README.md

```diff
@@ -93,72 +73,115 @@ npm install
 
 ## Usage
 
-### Generating a new key pair
+### Installing Support for Key Types
 
-Ed25519:
+In order to use this library, you will need to import and install driver
+libraries for key types you'll be working with via the `use()` method.
 
-```js
-const {Ed25519KeyPair} = require('crypto-ld');
+For example, to use this library with only the `ed25519` key type:
 
-const keyPair = await Ed25519KeyPair.generate();
 ```
+const {cryptoLd} = require('crypto-ld');
+cryptoLd.use(require('ed25519-key-pair');
 
-RSA:
+// With only one key type installed, you do not need to specify key type for
+// most operations
+const keyPair = await cryptoLd.generate(); // generates an ed25519 key pair
+```
 
-```js
-const {RSAKeyPair} = require('crypto-ld');
+To use the library with all supported key types:
 
-const keyPair = await RSAKeyPair.generate();
 ```
+const {cryptoLd} = require('crypto-ld');
+cryptoLd.use(require('ed25519-key-pair')); // ed25519 type
+cryptoLd.use(require('rsa-key-pair')); // rsa type
+cryptoLd.use(require('secp256k1-key-pair')); // secp256k1 type
+cryptoLd.use(require('x25519-key-pair')); // x25519 type
+
+// When using multiple key types, you'll need to specify type when generating
+const edKeyPair = await cryptoLd.generate({type: 'ed25519'});
+const rsaKeyPair = await cryptoLd.generate({type: 'rsa'});
+```
+
+### Generating a new public/private key pair
+
+To generate a new public/private key pair: `cryptoLd.generate(options)`:
 
-### Exporting a public/private key pair
+* `{string} [type]` Optional if only one key type is installed, required otherwise. 
+* `{string} [controller]` Optional controller URI or DID to initialize the
+  generated key. (This will also init the key id.) 
+* `{string} [seed]` Optional deterministic seed value (only supported by some
+  key types, such as `ed25519`) from which to generate the key.
+
+### Importing a key pair from storage
+
+To create an instance of a public/private key pair from data imported from
+storage, use `cryptoLd.from()`:
 
 ```js
-const edKeyPair = await Ed25519KeyPair.generate();
-edKeyPair.id = 'did:ex:123#' + edKeyPair.fingerprint();
+const serializedKeyPair = await loadFromSomeStorage();
+
+const keyPair = cryptoLd.from(serializedKeyPair);
+```
+
+Note that only installed key types are supported, if you try to create a
+key pair via `from()` for an unsupported type, an error will be thrown.
+
+### Common individual key pair operations
+
+The full range of operations will depend on key type. Here are some common
+operations supported by all key types.
 
-console.log(await edKeyPair.export())
-/* ->
+#### Exporting the public key only
+
+To export just the public key of a pair - `exportPublic()`:
+
+```
+await keyPair.exportPublic();
```

Comment:
I believe we should make this just `export()`. The proposed API change should be.

```js
// export just public key data
await keyPair.export(); // same as await keyPair.export({includePrivate: false})
// export public and private key data
await keyPair.export({includePrivate: true})

```

Reply:
I would like to counter-propose that we leave it as `exportPublic()`. When reading code, when I see `keyPair.export()`, I do not know if that's supposed to include or exclude the public key without going to look up the function definition.

Reply:
That's fine. Why not update the API to the following:

```js
// export just public key data
await keyPair.exportPublic(); // same as await keyPair.exportPublic({includePrivate: false})
// export public and private key data
await keyPair.exportPublic({includePrivate: true})

```

Reply:
-1 to having a function that says it's doing X do something else because of an input parameter. In this case `X` is "export the public key" and "something else" is "export the private key". Either the function should be called `export` and what it exports depends on the input parameters (we decide that @dmitrizagidulin's concern about looking up the definition isn't a good enough reason to change the name), or we have two functions as proposed by @dmitrizagidulin.

At first glance I think I'd prefer the latter, but I have to say -- I'm going to have to look up what `export`/`exportPublic` returns so I have to look up the definition anyway! If we want to address @dmitrizagidulin's concern about having to look up the definition, perhaps a better solution is that you must always specify whether the private key is included and you must also specify the format you want as output. Then anyone reading it does not need to look up the definition. Otherwise, they will have to. It seems naming it `export` and being explicit with the parameters is best given this rationale.

Reply:
> It seems naming it `export` and being explicit with the parameters is best given this rationale.

@dlongley Agreed. My preference is using `export` rather than `exportPublic`. The latter was proposed as a compromise to @dmitrizagidulin's counter proposal. The point you bring up, having to look at `export`/`exportPublic` return values anyways is the point. Splitting up the functions does not alleviate the need to look up function definitions and now we have two functions that have "export" functionality. It seems clear to have a single `export` with explicit parameters.

Reply:
@gannan08 ok, sounds good. I like this proposal, of a single `export()` with explicit params. Will refactor.

Reply:
@gannan08 @dlongley I changed it to just have one function, `export()`, as requested.